File: qyran/lib/api/entity/UserEntity.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:qyran/api/entity/enums/UserRole.dart';

part 'UserEntity.g.dart';

@JsonSerializable()
class UserEntity {
  String id;
  String phone;
  String name;
  String surname;
  String email;
  String? social_1;
  String? social_2;
  UserRole role;

  UserEntity({required this.id, required this.phone, required this.name, required this.surname, required this.email, required this.role, this.social_1, this.social_2});

  factory UserEntity.fromJson(Map<String, dynamic> json) => _$UserEntityFromJson(json);
  Map<String, dynamic> toJson() => _$UserEntityToJson(this);
}

File: server/src/main/java/com/thedeveloper/qyran/entity/UserEntity.java
package com.thedeveloper.qyran.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.thedeveloper.qyran.enums.UserRole;
import jakarta.persistence.*;
import lombok.Data;

import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "users")
@Data
public class UserEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    String id;
    @Column(unique = true)
    String phone;
    @JsonIgnore
    String password;
    String name;
    String surname;
    String email;
    String social_1;
    String social_2;
    @JsonIgnore
    String photo;
    @Enumerated(EnumType.STRING)
    UserRole role;
    @ManyToMany(fetch = FetchType.EAGER)
    @JsonIgnore
    List<VideoEntity> videoListView = new ArrayList<>();
    @JsonIgnore
    @ManyToMany(fetch = FetchType.EAGER)
    List<TestEntity> testListView = new ArrayList<>();
    @ManyToMany(fetch = FetchType.EAGER)
    @JsonIgnore
    List<CourseEntity> courseList = new ArrayList<>();
}
